internal/collector/signal.go

```diff
@@ -11,10 +11,32 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/hashicorp/go-version"
 	"github.com/sirupsen/logrus"
 )
 
-func sigNum(sig string) int {
+func isSigNumSupport() bool {
+	v := keepalivedVersion()
+	sigNumSupportedVersion, err := version.NewVersion("1.3.8")
+	if err != nil {
+		logrus.WithError(err).Fatal("Unexcpected error")
+	}
+
+	return v.GreaterThanOrEqual(sigNumSupportedVersion)
+}
+
+func sigNum(sig string) os.Signal {
+	if !isSigNumSupport() {
+		switch sig {
+		case "DATA":
+			return syscall.SIGUSR1
+		case "STATS":
+			return syscall.SIGUSR2
+		default:
+			logrus.WithField("signal", sig).Fatal("Unsupported signal for your keepalived")
```

Comment:
signum is calling while creating `KeepalivedCollector` instance. because of this all errors occurs here would be "FATAL" to this app like other errors in that function.

Reply:
It's okay then.